ompi/communicator/comm.c

```diff
@@ -1581,11 +1583,11 @@ int ompi_comm_get_rprocs ( ompi_communicator_t *local_comm,
         rc = MCA_PML_CALL(send (&int_len, 1, MPI_INT, remote_leader, tag,
                                 MCA_PML_BASE_SEND_STANDARD, bridge_comm ));
         if ( OMPI_SUCCESS != rc ) {
-            goto err_exit;
+            rlen = 0;  /* complete the recv and then the collectives */
         }
         rc = ompi_request_wait( &req, MPI_STATUS_IGNORE );
         if ( OMPI_SUCCESS != rc ) {
-            goto err_exit;
+            rlen = 0;  /* participate in the collective and then done */
```

Comment:
This changes the behavior without FT too, but I guess it doesn't matter because this happens only when there is an error?

Reply:
Yes, this could deadlock for non-FT related errors just as well.